oonib/report/handlers.py

```diff
@@ -101,24 +114,23 @@ def parseUpdateReportRequest(request):
     # We are also keeping in memory multiple copies of the same object. A lot
     # of optimization can be done.
     parsed_request = json.loads(request)
-    try:
-        report_id = parsed_request['report_id']
-    except KeyError:
+
+    report_id = parsed_request.get('report_id', report_id)
+    if not report_id:
         raise e.MissingField('report_id')
 
     if not re.match(report_id_regexp, report_id):
         raise e.InvalidRequestField('report_id')
 
+    parsed_request['report_id'] = report_id
```

Comment:
Here, aren't you overwriting `report_id` with `report_id`? That is, you have excluded the case in which the report id read from `parsed_request` was not set and the one in which was invalid, so you probably know it is valid and present at this point, so why you overwrite it?